src/App.tsx

```diff
@@ -51,9 +68,45 @@ function generateHue() {
   return Math.floor(Math.random() * 360);
 }
 
+/** 
+ * This appears to still be the best way to download a file while suggesting a filename.
+ * 
+ * According to [mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download) 
+ * the download attribute gets ignored on URIs that are not either `same-origin` or use the `blob` or `data` schemes.
+ */
+function downloadUri(uri: string, filename: string) {
+  const downloadAnchor = document.createElement("a");
+  console.log(filename);
+  downloadAnchor.download = filename;
+  downloadAnchor.href = uri;
+  downloadAnchor.click();
+}
+
+async function eventToAsync<EventType extends keyof HTMLElementEventMap>(element: HTMLElement, eventType: EventType) {
+  const abortController = new AbortController();
+  const evt = await new Promise<HTMLElementEventMap[EventType]>((resolve) => {
+    element.addEventListener(eventType, (evt) => resolve(evt), {signal: abortController.signal});
+  })
+  abortController.abort();
+  return evt;
+}
+
+/**
+ * This appears to be the best way to upload a file.
+ */
+async function uploadFile() {
+  const uploadInput = document.createElement("input");
+  uploadInput.type = "file";
+  uploadInput.click();
+  await eventToAsync(uploadInput, "change");
+  const files = uploadInput.files;
+  if (files?.length !== 1) return;
```

Comment:
Is this something that can actually happen? If not, would be useful to note that in the code. If it is, maybe show an error message?

Reply:
It should not. With how I call this dialog it should require exactly one selected File.

But I'm not sure how compliant all browsers are to that.

I'm also unsure if all browsers react the same to the user aborting the file selection. And also how we should react in this case btw. This might actually be a resource leak here (not THAT relevant, its unlikely a user opens more than a dozen of these just to abort them, before reloading the page for one or another reason). All these behaviors seem not well documented and probably are not consistent.

I could create a message for this case, Its just not unlikely we wont cover all bases with it or even this guard clause.

If you want a message, did you mean an easy to spot one with a toast, or a console.error or similar?